extensions/hazelcast-client/runtime/pom.xml

```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-hazelcast-client-parent</artifactId>
+        <version>999-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>quarkus-hazelcast-client</artifactId>
+    <name>Quarkus - Hazelcast - Client - Runtime</name>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-arc</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-jaxb</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.hazelcast</groupId>
+            <artifactId>hazelcast-all</artifactId>
+            <version>3.12.5</version>
+        </dependency>
+        <dependency>
+            <groupId>javax.cache</groupId>
+            <artifactId>cache-api</artifactId>
+            <version>1.1.1</version>
```

Comment:
Same here it should be in the BOM. But it doesn't come with Hazelcast? I'm surprised you even need to add it.

I'm also totally unsure it's useful as we don't have JSR 107 support in Quarkus.

Reply:
Hazelcast 3.0.0 has `cache-api` in as _provided_ dependency - most likely we should be able to get rid of it in the next release where we're reshaping our artifacts

Reply:
Is this dependency still needed now that Hazelcast Client 4.0 is used?

Reply:
Unfortunately, yes. The dependency is not used for anything other than making GraalVM happy, and manual removal of it from core classes via substitutions would last weeks and be super fragile.

The real hope to get rid of is if this gets implemented: https://github.com/oracle/graal/issues/1664 and then similar feature could be exposed through Quarkus build item

Reply:
Are you sure that stripping out the classes via substitutions would be that difficult? We really try to minimize runtime dependencies as much as possible.

Reply:
To give you a clue what the volume of changes is: https://github.com/hazelcast/hazelcast/search?q=javax.cache&unscoped_q=javax.cache

We're totally coupled to jcache

Reply:
I see, thanks

Reply:
What we could do instead is to make the dependency optional and include a note in the README that it's required for native... let me know if this is ok

Reply:
I think it's probably better as is. We really try to make the switch from jvm mode to native just a matter of adding `-Dnative`.